src/main/java/mate/academy/dao/impl/BookDaoImpl.java

```diff
@@ -0,0 +1,155 @@
+                    throw new DataProcessingException(
+                            "Failed to insert new book record into the database", e);
+                }
+            }
+            if (statement != null) {
+                try {
+                    statement.close();
+                } catch (SQLException e) {
+                    throw new DataProcessingException(
+                            "Failed to close PreparedStatement after attempting "
+                                    + "to insert new book", e);
+                }
+            }
+            if (connection != null) {
+                try {
+                    connection.close();
+                } catch (SQLException e) {
+                    throw new DataProcessingException(
+                            "Failed to close Connection after attempting "
+                                    + "to insert new book", e);
+                }
+            }
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT * FROM books WHERE id = ?";
+        Optional<Book> result = Optional.empty();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement =
+                        connection.prepareStatement(sql)) {
+
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+
+            if (resultSet.next()) {
+                Book book = new Book();
+                book.setId(resultSet.getObject("id", Long.class));
+                book.setTitle(resultSet.getString("title"));
+                book.setPrice(resultSet.getObject("price", BigDecimal.class));
+                return Optional.of(book);
+            }
+
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Failed to retrieve book from the database for ID: " + id, e
+            );
+        }
+        return result;
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> bookList = new ArrayList<>();
+        String sql = "SELECT * FROM books";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement =
+                        connection.prepareStatement(sql)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Book book = new Book();
+                book.setId(resultSet.getObject("id", Long.class));
+                book.setTitle(resultSet.getString("title"));
+                book.setPrice(resultSet.getObject("price", BigDecimal.class));
+                bookList.add(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Failed to retrieve books from the database", e
+            );
+        }
+        return bookList;
+    }
+
+    @Override
+    public Book update(Book book) {
```

Comment:
Why this method return null?

Reply:
I am sorry, but I'm confused and don't understand which comment I didn't correct. Please tell me specifically what I need to fix. Thank you very much!

Reply:
Sure. 
I noticed that you made a change - return book in update method. Good job!
Also, you are using `statement.executeUpdate();` in update method without checking if any errors appear. That's why I recommended you to add this changes that you didn't fixed:

```
int affectedRows = statement.executeUpdate();
if (affectedRows < 1) {...
```

In addition, you have a lot of blank lines in findById method that i also mentioned in the comments (but this is less important).